weaviate/collections/batch/base.py

```diff
@@ -184,10 +176,14 @@ def _add_failed_objects_from_response(
         # return successful_responses
 
 
-class _Batch:
+B = TypeVar("B", bound="_BatchBase")
```

Comment:
Is there any upside of this over using "_BatchBase" directly as type? Besides having less strings?

Reply:
Yes, because the function signatures of `_BatchClient` and `_BatchCollection` are different (one has `collection`, the other doesn't). In order to not violate LSP, these functions cannot be defined abstractly on a base class and inherited (since their signatures are different). However, we want the `__enter__` method to return the correct type when the user does `with client.batch as batch` and `with collection.batch as batch`, so we must define a type bound by `_BatchBase` that achieves this exact outcome

Reply:
thanks!